docs/dev_guide/new_objects.rst

```diff
@@ -4,6 +4,37 @@
 Creating new SunPy Subclasses (Maps, TimeSeries)
 ************************************************
 
+Documenting new data sources
+----------------------------
+Each subclass of `~sunpy.map.GenericMap` or `~sunpy.timeseries.GenericTimeSeries`
+must provide a detailed docstring providing an overview of the data source that
+the object represents. In order to maintain consistency and completeness the
+following information must be provided by a data source docstring, if available,
+and preferably in the following order:
+
+* the name of the mission and instrument and the institution that built it
+* a short description of the instrument (e.g. Cassegrain reflector, Wolter-1 grazing incidence x-ray, coronagraph) including the type of detector
+* a description of the platform (e.g. satellite in 28 deg inclined orbit, a telescope on the summit of Mauna Kea in Hawaii)
+* a description of the primary purpose or science goals of the instrument.
```

Comment:
Science goals seems like a very broad/vague requirement - much better to link to an instrument paper that provides this information.

Reply:
I was thinking that providing a single sentence summarizing what the instrument was designed to do would be useful in most cases.